modulemd/modulemd-build-config.c

```diff
@@ -364,6 +364,56 @@ modulemd_build_config_parse_yaml (yaml_parser_t *parser,
 }
 
 
+gboolean
+modulemd_build_config_emit_yaml (ModulemdBuildConfig *self,
+                                 yaml_emitter_t *emitter,
+                                 GError **error)
+{
+  MODULEMD_INIT_TRACE ();
+  int ret;
+  g_autoptr (GError) nested_error = NULL;
+  MMD_INIT_YAML_EVENT (event);
+
+  ret = mmd_emitter_start_mapping (
+    emitter, YAML_BLOCK_MAPPING_STYLE, &nested_error);
```

Comment:
Before we actually start emitting content, please run `modulemd_build_config_validate()` to ensure that the contents are acceptable to output.

Reply:
Shouldn't the validation be done just before starting to emit content of any higher-level stand-alone document format that makes use of BuildConfig? Such upper level validation should already be calling `modulemd_build_config_validate()`.

Reply:
Oh right. You are correct of course.